model/src/main/kotlin/utils/FileArchiver.kt

```diff
@@ -125,4 +133,15 @@ class FileArchiver(
             logger.error { "Failed to unarchive data for $provenance: ${it.collectMessages()}" }
         }.isSuccess
     }
+
+    internal fun File.isValidTextFile(): Boolean {
+        val detected = tika.detect(this)
+        val invalidTypes = listOf(
```

Comment:
Maybe make this list a top-level constant, so that it does not have to be recreated every time this function is called.